### PR TITLE
Support sending messages before peer connects

### LINKED ISSUE #3320: [Bug]: occasional NullPointerException on local WebRTC client connection
### Describe the Bug

After starting a WebRTC server it failed to start properly.

```
18:27:52.457 (ClientHandshake.java:145) [Thread-16] INFO  - skullman :send: CLIENT_INIT_MSG
Exception in thread "Thread-16" java.lang.NullPointerException: Cannot invoke "dev.onvoid.webrtc.RTCPeerConnection.getConnectionState()" because "this.peerConnection" is null
        at net.rptools.clientserver.simple.client.WebRTCClientConnection.sendMessage(WebRTCClientConnection.java:185)
        at net.rptools.clientserver.simple.client.WebRTCClientConnection.sendMessage(WebRTCClientConnection.java:178)
        at net.rptools.clientserver.hessian.client.MethodClientDecorator.sendMessage(MethodClientDecorator.java:42)
        at net.rptools.maptool.server.ClientHandshake.sendMessage(ClientHandshake.java:146)
        at net.rptools.maptool.server.ClientHandshake.startHandshake(ClientHandshake.java:139)
        at net.rptools.maptool.client.MapToolConnection.start(MapToolConnection.java:76)
        at net.rptools.maptool.client.MapTool.createConnection(MapTool.java:1227)
        at net.rptools.maptool.client.AppActions$62.lambda$executeAction$1(AppActions.java:2326)
        at net.rptools.maptool.client.AppActions$ClientAction.lambda$runBackground$0(AppActions.java:3496)
        at java.base/java.lang.Thread.run(Thread.java:831)
18:27:52.546 (WebRTCServerConnection.java:70) [WebSocketConnectReadThread-81] INFO  - S WebSocket connected

18:27:52.762 (WebRTCClientConnection.java:136) [WebSocketConnectReadThread-84] INFO  - C WebSocket connected

18:27:52.922 (WebRTCClientConnection.java:408) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - C PeerConnection.onRenegotiationNeeded
18:27:52.926 (WebRTCClientConnection.java:305) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - C PeerConnection.onSignalingChange: HAVE_LOCAL_OFFER
18:27:52.936 (WebRTCClientConnection.java:344) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - C PeerConnection.onIceGatheringChange GATHERING
[sentry-pool-1-thread-2] WARN io.sentry.connection.AbstractConnection.lockdown - Initiated a temporary lockdown because of exception: {"detail":"bad sentry DSN public key"}
[sentry-pool-1-thread-2] ERROR io.sentry.connection.AsyncConnection - An exception occurred while sending the event to Sentry.
io.sentry.connection.ConnectionException: {"detail":"bad sentry DSN public key"}
        at io.sentry.connection.HttpConnection.doSend(HttpConnection.java:216)
        at io.sentry.connection.AbstractConnection.send(AbstractConnection.java:88)
        at io.sentry.connection.AsyncConnection$EventSubmitter.run(AsyncConnection.java:187)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at java.base/java.lang.Thread.run(Thread.java:831)
Caused by: java.io.IOException: Server returned HTTP response code: 400 for URL: https://sentry.io/api/1404248/store/
        at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1985)
        at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1577)
        at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:224)
        at io.sentry.connection.HttpConnection.doSend(HttpConnection.java:172)
        ... 5 more
18:27:53.094 (WebRTCClientConnection.java:305) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - S PeerConnection.onSignalingChange: HAVE_REMOTE_OFFER
18:27:53.096 (WebRTCClientConnection.java:73) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - S  setRemoteDescription success.
18:27:53.098 (WebRTCClientConnection.java:305) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - S PeerConnection.onSignalingChange: STABLE
18:27:53.100 (WebRTCClientConnection.java:344) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - S PeerConnection.onIceGatheringChange GATHERING
18:27:53.240 (WebRTCClientConnection.java:305) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - C PeerConnection.onSignalingChange: STABLE
18:27:53.240 (WebRTCClientConnection.java:329) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - C PeerConnection.onIceConnectionChange CHECKING
18:27:53.241 (WebRTCClientConnection.java:243) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - C  setRemoteDescription success.
18:27:53.256 (WebRTCClientConnection.java:310) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - C PeerConnection.onConnectionChange CONNECTING
18:27:53.256 (WebRTCClientConnection.java:310) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - S PeerConnection.onConnectionChange CONNECTING
18:27:53.257 (WebRTCClientConnection.java:344) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - C PeerConnection.onIceGatheringChange COMPLETE
18:27:53.257 (WebRTCClientConnection.java:329) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - S PeerConnection.onIceConnectionChange CHECKING
18:27:53.257 (WebRTCClientConnection.java:344) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - S PeerConnection.onIceGatheringChange COMPLETE
18:27:53.258 (WebRTCClientConnection.java:310) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - C PeerConnection.onConnectionChange CONNECTED
18:27:53.258 (WebRTCClientConnection.java:310) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - S PeerConnection.onConnectionChange CONNECTED
18:27:53.258 (WebRTCClientConnection.java:329) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - C PeerConnection.onIceConnectionChange CONNECTED
18:27:53.258 (WebRTCClientConnection.java:329) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - S PeerConnection.onIceConnectionChange CONNECTED
18:27:53.258 (WebRTCClientConnection.java:329) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - C PeerConnection.onIceConnectionChange COMPLETED
18:27:53.261 (WebRTCClientConnection.java:440) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - C localDataChannel onStateChange OPEN
18:27:53.261 (WebRTCClientConnection.java:395) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - S PeerConnection.onDataChannel
18:27:53.262 (WebRTCClientConnection.java:440) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - S localDataChannel onStateChange OPEN
18:27:53.264 (ServerHandshake.java:212) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - skullman@Draconis :got: CLIENT_INIT_MSG
18:27:53.266 (ServerHandshake.java:202) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - skullman@Draconis :send: USE_AUTH_TYPE_MSG
18:27:53.267 (ClientHandshake.java:155) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - skullman :got: USE_AUTH_TYPE_MSG
18:27:53.271 (ClientHandshake.java:145) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - skullman :send: CLIENT_AUTH_MESSAGE
18:27:53.272 (ServerHandshake.java:212) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - skullman@Draconis :got: CLIENT_AUTH_MESSAGE
18:27:53.272 (ServerHandshake.java:202) [WebRTCClientConnection.WebRTCThread_skullman@Draconis] INFO  - skullman@Draconis :send: CONNECTION_SUCCESSFUL_MSG
18:27:53.277 (ClientHandshake.java:155) [WebRTCClientConnection.WebRTCThread_skullman] INFO  - skullman :got: CONNECTION_SUCCESSFUL_MSG
```

### To Reproduce

Start a WebRTC server a bunch of times and get unlucky.

### Expected Behaviour

Local client is always able to send messages.

### Screenshots

_No response_

### MapTool Info

1.11.4 with WebRTC patch

### Desktop

Linux[Debian]

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by fishface60**
The local client may send messages before logging into the signalling
server and beginning WebRTC peer connection establishment
since it already knows it exists and is reachable.

Since the client will also wake up the sendThread on connection state
change it should be safe to enqueue the message without notifying the
thread.

### Description of the Change

Add a null reference check to message sending, so that if logic changes elsewhere cause messages to be sent before the connection is completely established, then they will be queued rather than raising an exception and breaking.

### Possible Drawbacks

More code to maintain and if it should never be possible to send messages it no longer fails so early where it may be easier to debug.

---
This change is [Reviewable](https://reviewable.io/reviews/rptools/maptool/3321)

### CODE REVIEW

**Comment by thelsing** [comment]
I think bumping the version of the webrtc libs should be enough. See https://github.com/devopvoid/webrtc-java/issues/35

**Comment by fishface60** [comment]
> I think bumping the version of the webrtc libs should be enough. See [devopvoid/webrtc-java#35](https://github.com/devopvoid/webrtc-java/issues/35) But merging this is also ok.

Thanks, I'm convinced I misdiagnosed the cause of the error and this wouldn't fix it and it just didn't present itself when I tested.
I'll reword this to be for more defensive programming against the possibility of messages being sent before the connection is established.

**Comment by thelsing** [comment]
We use currently 0.3 in MT of the webrtc lib. So you your fix is correct. But I think it would be a good idea to bump the version to 0.5 too.

**Comment by Phergus** [comment]
@thelsing so you're good with this change?

**Review by thelsing — APPROVED** [review]
